src/modules/flow/http-server/http-server.c

```diff
@@ -0,0 +1,626 @@
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <time.h>
+
+#include "http-server-gen.h"
+#include "sol-flow.h"
+#include "sol-http.h"
+#include "sol-http-server.h"
+#include "sol-mainloop.h"
+#include "sol-util.h"
+#include "sol-vector.h"
+#include "sol_config.h"
+
+#define HTTP_HEADER_ACCEPT "Accept"
+#define HTTP_HEADER_CONTENT_TYPE "Content-Type"
+#define HTTP_HEADER_CONTENT_TYPE_TEXT "text/plain"
+#define HTTP_HEADER_CONTENT_TYPE_JSON "application/json"
+
+struct http_data {
+    struct sol_flow_node *node;
+    union {
+        struct sol_irange i;
+        struct sol_drange d;
+        bool b;
+        char *s;
+    } value;
+
+    char *path;
+};
+
+static struct sol_http_server *server = NULL;
+static int init_count = 0;
+
+static int
+start_server(struct http_data *http, const char *path,
+    int (*handler)(void *data, struct sol_http_request *request))
+{
+    int r;
+
+    if (!server) {
+        server = sol_http_server_new(HTTP_SERVER_PORT);
+        SOL_NULL_CHECK(server, -1);
+    }
+
+    init_count++;
+
+    http->path = strdup(path);
+    SOL_NULL_CHECK_GOTO(http->path, err);
+
+    r = sol_http_server_register_handler(server, http->path, handler, http);
+    SOL_INT_CHECK_GOTO(r, < 0, err_handler);
+
+    return 0;
+
+err_handler:
+    free(http->path);
+err:
+    init_count--;
+    if (!init_count) {
+        sol_http_server_del(server);
+        server = NULL;
+    }
+    return -1;
+}
+
+static void
+stop_server(struct http_data *http)
+{
+    sol_http_server_unregister_handler(server, http->path);
+    free(http->path);
```

Comment:
nope, this is relative to the node and not the server.
The node will stop handle this path even if the server will keep alive.